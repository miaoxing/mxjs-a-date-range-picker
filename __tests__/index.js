import 'core-js/stable';
import 'regenerator-runtime/runtime';

import React from 'react';
import {render} from '@testing-library/react';
import {Form, FormItem} from '@mxjs/a-form';
import {MemoryRouter} from 'react-router';
import {createPromise} from '@mxjs/test';
import $ from 'miaoxing';
import DateRangePicker from '..';

describe('date-range-picker', () => {
  test('date', async () => {
    const promise = createPromise();
    const promise2 = createPromise();
    $.http = jest.fn().mockImplementation(() => promise.resolve({
      code: 1,
      message: 'success',
      data: {
        startDate: '2020-01-01',
        endDate: '2020-02-02',
      }
    }));

    const {container, findByDisplayValue} = render(<MemoryRouter>
      <Form>
        <FormItem name="_date">
          <DateRangePicker names={['startDate', 'endDate']}/>
        </FormItem>
      </Form>
    </MemoryRouter>);

    await promise;
    await findByDisplayValue('2020-01-01');
    expect(container.querySelector('#_date').value).toBe('2020-01-01');

    $.http.mockImplementation(() => promise2.resolve({
      code: 1,
      message: 'success',
      data: {}
    }));

    container.querySelector('form').submit();
    await promise2;

    expect($.http).toMatchSnapshot();
  });

  test('time', async () => {
    const promise = createPromise();
    const promise2 = createPromise();
    $.http = jest.fn().mockImplementation(() => promise.resolve({
      code: 1,
      message: 'success',
      data: {
        startDate: '2020-01-01 01:01:01',
        endDate: '2020-02-02 02:02:02',
      }
    }));

    const {container, findByDisplayValue} = render(<MemoryRouter>
      <Form>
        <FormItem name="_date">
          <DateRangePicker names={['startDate', 'endDate']} showTime/>
        </FormItem>
      </Form>
    </MemoryRouter>);

    await promise;
    await findByDisplayValue('2020-01-01 01:01:01');
    expect(container.querySelector('#_date').value).toBe('2020-01-01 01:01:01');

    $.http.mockImplementation(() => promise2.resolve({
      code: 1,
      message: 'success',
      data: {}
    }));

    container.querySelector('form').submit();
    await promise2;

    expect($.http).toMatchSnapshot();
  });

  test('only start date', async() => {
    const promise = createPromise();
    const promise2 = createPromise();
    $.http = jest.fn().mockImplementation(() => promise.resolve({
      code: 1,
      message: 'success',
      data: {
        startDate: '2020-01-01',
        endDate: '',
      }
    }));

    const {container, findByDisplayValue} = render(<MemoryRouter>
      <Form>
        <FormItem name="_date">
          <DateRangePicker names={['startDate', 'endDate']}/>
        </FormItem>
      </Form>
    </MemoryRouter>);

    await promise;
    await findByDisplayValue('2020-01-01');
    expect(container.querySelector('#_date').value).toBe('2020-01-01');

    $.http.mockImplementation(() => promise2.resolve({
      code: 1,
      message: 'success',
      data: {}
    }));

    container.querySelector('form').submit();
    await promise2;

    expect($.http).toMatchSnapshot();
  });

  test('empty value', async() => {
    const promise = createPromise();
    const promise2 = createPromise();
    $.http = jest.fn().mockImplementation(() => promise.resolve({
      code: 1,
      message: 'success',
      data: {}
    }));

    const {container, findByDisplayValue} = render(<MemoryRouter>
      <Form>
        <FormItem name="_date">
          <DateRangePicker names={['startDate', 'endDate']}/>
        </FormItem>
      </Form>
    </MemoryRouter>);

    await promise;

    $.http.mockImplementation(() => promise2.resolve({
      code: 1,
      message: 'success',
      data: {}
    }));

    container.querySelector('form').submit();
    await promise2;

    expect($.http).toMatchSnapshot();
  });

  test('custom format', async () => {
    const promise = createPromise();
    const promise2 = createPromise();
    $.http = jest.fn().mockImplementation(() => promise.resolve({
      code: 1,
      message: 'success',
      data: {
        startDate: '2020-01-01',
        endDate: '',
      }
    }));

    const {container, findByDisplayValue} = render(<MemoryRouter>
      <Form>
        <FormItem name="_date">
          <DateRangePicker names={['startDate', 'endDate']} format="YYYY/MM/DD"/>
        </FormItem>
      </Form>
    </MemoryRouter>);

    await promise;
    await findByDisplayValue('2020/01/01');
    expect(container.querySelector('#_date').value).toBe('2020/01/01');

    $.http.mockImplementation(() => promise2.resolve({
      code: 1,
      message: 'success',
      data: {}
    }));

    container.querySelector('form').submit();
    await promise2;

    expect($.http).toMatchSnapshot();
  });
});
