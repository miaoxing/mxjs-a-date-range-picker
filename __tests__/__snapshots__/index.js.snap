// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`date-range-picker custom format 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "method": "get",
        "url": "",
      },
    ],
    Array [
      Object {
        "data": Object {
          "endDate": "",
          "startDate": "2020/01/01",
        },
        "loading": true,
        "method": "POST",
        "url": "",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {
        "reject": [Function],
        "resolve": [Function],
      },
    },
    Object {
      "type": "return",
      "value": Promise {
        "reject": [Function],
        "resolve": [Function],
      },
    },
  ],
}
`;

exports[`date-range-picker date 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "method": "get",
        "url": "",
      },
    ],
    Array [
      Object {
        "data": Object {
          "endDate": "2020-02-02",
          "startDate": "2020-01-01",
        },
        "loading": true,
        "method": "POST",
        "url": "",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {
        "reject": [Function],
        "resolve": [Function],
      },
    },
    Object {
      "type": "return",
      "value": Promise {
        "reject": [Function],
        "resolve": [Function],
      },
    },
  ],
}
`;

exports[`date-range-picker empty value 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "method": "get",
        "url": "",
      },
    ],
    Array [
      Object {
        "data": Object {
          "endDate": "",
          "startDate": "",
        },
        "loading": true,
        "method": "POST",
        "url": "",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {
        "reject": [Function],
        "resolve": [Function],
      },
    },
    Object {
      "type": "return",
      "value": Promise {
        "reject": [Function],
        "resolve": [Function],
      },
    },
  ],
}
`;

exports[`date-range-picker only start date 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "method": "get",
        "url": "",
      },
    ],
    Array [
      Object {
        "data": Object {
          "endDate": "",
          "startDate": "2020-01-01",
        },
        "loading": true,
        "method": "POST",
        "url": "",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {
        "reject": [Function],
        "resolve": [Function],
      },
    },
    Object {
      "type": "return",
      "value": Promise {
        "reject": [Function],
        "resolve": [Function],
      },
    },
  ],
}
`;

exports[`date-range-picker time 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "method": "get",
        "url": "",
      },
    ],
    Array [
      Object {
        "data": Object {
          "endDate": "2020-02-02 02:02:02",
          "startDate": "2020-01-01 01:01:01",
        },
        "loading": true,
        "method": "POST",
        "url": "",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {
        "reject": [Function],
        "resolve": [Function],
      },
    },
    Object {
      "type": "return",
      "value": Promise {
        "reject": [Function],
        "resolve": [Function],
      },
    },
  ],
}
`;
